features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemImageView.kt

```diff
@@ -86,13 +88,23 @@ fun TimelineItemImageView(
                     modifier = Modifier
                         .fillMaxWidth()
                         .then(if (isLoaded) Modifier.background(Color.White) else Modifier),
-                    model = MediaRequestData(
-                        source = content.preferredMediaSource,
-                        kind = MediaRequestData.Kind.File(
-                            fileName = content.filename,
-                            mimeType = content.mimeType,
-                        ),
-                    ),
+                    model = if (content.thumbnailSource != null) {
+                        MediaRequestData(
+                            source = content.thumbnailSource,
+                            kind = MediaRequestData.Kind.Thumbnail(
+                                width = content.thumbnailWidth?.toLong() ?: MAX_THUMBNAIL_WIDTH,
+                                height = content.thumbnailHeight?.toLong() ?: MAX_THUMBNAIL_HEIGHT,
+                            ),
+                        )
+                    } else {
+                        MediaRequestData(
+                            source = content.thumbnailSource,
```

Comment:
I think we should keep `content.preferredMediaSource` and always try to use `kind = MediaRequestData.Kind.Thumbnail`

Reply:
Also, the thumbnailAPI is probably not working with GIF?

Reply:
Nope, it doesn't work properly... I guess we'll need to check the mime type here too.

Reply:
This is already done in `preferredMediaSource`, so maybe we could have a `mediaRequestData` extension function in `Content`, to pick the correct `source` and `kind`?

Reply:
I tried something like that in https://github.com/element-hq/element-x-android/pull/3817/commits/6da362e499428bbc3fd68eaf8bce57f21d8e3a2c.

Reply:
But why do you want to use `Kind.File` when using `mediaSource` (except for isMimeTypeAnimatedImage I mean).
I'd have done something like that : 

```
val thumbnailMediaRequest: MediaRequestData by lazy {
        if (mimeType.isMimeTypeAnimatedImage()) {
            MediaRequestData(
                source = mediaSource,
                kind = MediaRequestData.Kind.File(
                    fileName = filename,
                    mimeType = mimeType
                )
            )
        } else {
            MediaRequestData(
                source = thumbnailSource ?: mediaSource,
                kind = MediaRequestData.Kind.Thumbnail(
                    width = thumbnailWidth?.toLong() ?: MAX_THUMBNAIL_WIDTH,
                    height = thumbnailHeight?.toLong() ?: MAX_THUMBNAIL_HEIGHT
                ),
            )
        }
    }
```

and removed the `preferredMediaSource`?

Reply:
Used this in https://github.com/element-hq/element-x-android/pull/3817/commits/daba49cac0b560f105ae8257595f457a3cb57e59.